network/src/protocols/health_checker/interface.rs

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    application::{
+        db::PeerDb,
+        interface::{NetworkInterface, PeerStateChange},
+        management::PeerManagementInterface,
+        types::PeerState,
+    },
+    error::NetworkError,
+    protocols::{
+        health_checker::{
+            HealthCheckerMsg, HealthCheckerNetworkEvents, HealthCheckerNetworkSender,
+        },
+        network::Event,
+    },
+};
+use async_trait::async_trait;
+use diem_types::PeerId;
+use futures::{stream::FusedStream, Stream};
+use std::{
+    pin::Pin,
+    sync::Arc,
+    task::{Context, Poll},
+};
+
+/// HealthChecker's view into networking
+pub struct HealthCheckNetworkInterface {
+    peer_db: Arc<PeerDb>,
+    sender: HealthCheckerNetworkSender,
+    receiver: HealthCheckerNetworkEvents,
+}
+
+impl HealthCheckNetworkInterface {
+    pub fn new(
+        peer_db: Arc<PeerDb>,
+        sender: HealthCheckerNetworkSender,
+        receiver: HealthCheckerNetworkEvents,
+    ) -> Self {
+        Self {
+            peer_db,
+            sender,
+            receiver,
+        }
+    }
+
+    /// Disconnect a peer, and keep track of the associated state
+    /// Note: This removes the peer outright for now until we add GCing, and historical state management
+    pub async fn disconnect_peer(&mut self, peer_id: PeerId) -> Result<(), NetworkError> {
+        // Possibly already disconnected, but try anyways
+        let _ = self.update_state(peer_id, PeerState::Disconnecting);
```

Comment:
Silly question: where do we set it to Disconnected? Or is it meant to go "Disconnecting" -> Removed entirely?

Reply:
Without a proper GC strategy, I'm just removing them entirely after going Disconnecting.

There's an option later to keep the data, and then remove it when it reaches a point, or when it's no longer in the trusted peer set.